packages/wm/src/commands/general/platform_sync.rs

```diff
@@ -257,9 +257,21 @@ fn redraw_containers(
       },
     );
 
-    let rect = window
+    let scale_factor = window
+      .monitor()
+      .and_then(|monitor| monitor.native().scale_factor().ok());
+
+    let mut rect = window
       .to_rect()?
-      .apply_delta(&window.total_border_delta()?, None);
+      .apply_delta(&window.total_border_delta()?, scale_factor);
+
+    // Constrain tiling windows to their workspace's bounds to prevent
+    // overflow on multi-monitor setup with different scaling factors.
+    if window.is_tiling_window() {
+      if let Some(workspace) = window.workspace() {
+        rect = rect.constrain_within(&workspace.to_rect()?);
+      }
+    }
```

Comment:
```suggestion
    if window.is_tiling_window() {
      rect = rect.constrain_within(&workspace.to_rect()?);
    }
```
window's workspace is already in scope 👍